esi/src/lib.rs

```diff
@@ -504,7 +517,97 @@ fn event_receiver(
                 except_task,
             });
         }
-        Event::XML(event) => {
+        Event::ESI(Tag::Assign { name, value }) => {
+            // TODO: the 'name' here might have a subfield, we need to parse it
+            let result = evaluate_expression(&value, ctx)?;
+            ctx.set_variable(&name, None, result);
+        }
+        Event::ESI(Tag::Vars { name }) => {
+            if let Some(name) = name {
+                let result = evaluate_expression(&name, ctx)?;
+                queue.push_back(Element::Raw(result.to_string().into_bytes()));
+            }
+        }
+        Event::ESI(Tag::When { .. }) => {
+            println!("Shouldn't be possible to get a When tag here");
+        }
+        Event::ESI(Tag::Choose {
+            when_branches,
+            otherwise_events,
+        }) => {
+            let mut chose_branch = false;
+            for (when, events) in when_branches {
+                if let Tag::When { test, match_name } = when {
+                    if let Some(match_name) = match_name {
+                        ctx.set_match_name(&match_name);
+                    }
+                    let result = evaluate_expression(&test, ctx)?;
+                    if result.to_bool() {
+                        chose_branch = true;
+                        for event in events {
+                            event_receiver(
+                                event,
+                                queue,
+                                is_escaped,
+                                original_request_metadata,
+                                dispatch_fragment_request,
+                                ctx,
+                            )?;
+                        }
+                        break;
+                    }
+                } else {
+                    println!("Somehow got something other than a When in a Choose: {when:?}",);
```

Comment:
Error?

Reply:
Should be unreachable, so marked it as such.